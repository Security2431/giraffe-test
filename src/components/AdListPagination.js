import React from 'react';
import PropTypes from 'prop-types';

class AdListPagination extends React.Component {
  static propTypes = {
    pageSize: PropTypes.number.isRequired,
    adsCount: PropTypes.number.isRequired,
    currentPage: PropTypes.func.isRequired,
  };

  state = {
    currentPage: 1,
  };

  componentDidMount() {
    // eslint-disable-next-line
    this.props.currentPage(this.state.currentPage);
  }

  componentDidUpdate() {
    const { currentPage } = this.state;
    const { pageSize, adsCount } = this.props;
    const pagesCount = Math.ceil(adsCount / pageSize);

    if (pagesCount && pagesCount < currentPage) {
      this.handleClick(pagesCount);
    }
  }

  handleClick = (index, e) => {
    // eslint-disable-next-line
    e && e.preventDefault && e.preventDefault();

    // eslint-disable-next-line
    this.props.currentPage(index);

    this.setState({
      currentPage: index,
    });
  };

  render() {
    const { currentPage } = this.state;
    const { pageSize, adsCount } = this.props;
    const pagesCount = Math.ceil(adsCount / pageSize);

    return pageSize < adsCount ? (
      <React.Fragment>
        <nav aria-label="Page navigation example">
          <ul className="pagination">
            <li className={`page-item ${currentPage <= 1 && 'disabled'}`}>
              <button
                className="page-link"
                type="button"
                aria-label="Previous"
                onClick={e => this.handleClick(currentPage - 1, e)}
              >
                <span aria-hidden="true">&laquo;</span>
                <span className="sr-only">Previous</span>
              </button>
            </li>

            {[...Array(pagesCount)].map((page, i) => (
              <li
                className={`page-item ${i + 1 === currentPage && 'active'}`}
                // eslint-disable-next-line
                key={i}
              >
                <button
                  className="page-link"
                  type="button"
                  onClick={e => this.handleClick(i + 1, e)}
                >
                  {i + 1}
                </button>
              </li>
            ))}

            <li className={`page-item ${currentPage >= pagesCount && 'disabled'}`}>
              <button
                className="page-link"
                type="button"
                aria-label="Next"
                onClick={e => this.handleClick(currentPage + 1, e)}
              >
                <span aria-hidden="true">&raquo;</span>
                <span className="sr-only">Next</span>
              </button>
            </li>
          </ul>
        </nav>
      </React.Fragment>
    ) : null;
  }
}

export default AdListPagination;
