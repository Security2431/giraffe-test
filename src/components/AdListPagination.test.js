/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AdListPagination from './AdListPagination';

const mockProps = {
  pageSize: 5,
  adsCount: 100,
  currentPage: jest.fn(),
};

describe('<AdListPagination />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AdListPagination {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AdListPagination {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
