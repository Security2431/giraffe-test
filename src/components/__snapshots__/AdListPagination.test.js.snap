// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdListPagination /> renders correctly 1`] = `
<nav
  aria-label="Page navigation example"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item disabled"
    >
      <button
        aria-label="Previous"
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          className="sr-only"
        >
          Previous
        </span>
      </button>
    </li>
    <li
      className="page-item active"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        10
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        11
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        12
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        13
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        14
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        15
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        16
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        17
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        18
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        19
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        20
      </button>
    </li>
    <li
      className="page-item false"
    >
      <button
        aria-label="Next"
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          className="sr-only"
        >
          Next
        </span>
      </button>
    </li>
  </ul>
</nav>
`;
